Library/Homebrew/dev-cmd/linkage.rb

```diff
@@ -23,22 +23,7 @@ def linkage
         result.display_test_output
         Homebrew.failed = true if result.broken_dylibs?
         if OS.linux?
-          host_whitelist = %w[
-            libc.so.6
-            libcrypt.so.1
-            libdl.so.2
-            libm.so.6
-            libnsl.so.1
-            libpthread.so.0
-            librt.so.1
-            libutil.so.1
-
-            libgcc_s.so.1
-            libgomp.so.1
-            libstdc++.so.6
-          ]
-          host_deps = result.system_dylibs.to_a.map { |s| File.basename s }
-          Homebrew.failed = true unless (host_deps - host_whitelist).empty?
+          Homebrew.failed = true if result.bad_system_dylibs?
```

Comment:
Drop the `if OS.linux?` guard.
Please rename `bad_system_dylibs?` to `unwanted_system_dylibs?`